subxt/src/backend/chain_head/follow_stream.rs

```diff
@@ -72,6 +72,10 @@ enum InnerStreamState<Hash> {
     ReceivingEvents(FollowEventStream<Hash>),
     /// We received a stop event. We'll send one on and restart the stream.
     Stopped,
+    /// The backend was closed and tell everything to shutdown.
+    BackendClosed,
+    /// Report error.
+    Error(Option<Error>),
```

Comment:
Offhand I can't see why the Error is in an `Option` here? Can't see anywhere it's set to `None` :D

Reply:
It's just hack around that poll takes &mut and not possible to clone the error.

Thus, I had to use Option::take to take the error and then change the state.

Reply:
Aah, yeah ok gotcha!

Cloning the error is important if we wanted to take the alternate approach of forwarding the error to any subscribers, so hmm I can see why that could be a pain!